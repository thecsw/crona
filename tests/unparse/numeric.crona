// Handles:
// int BinaryExpNode subclasses
// int UnaryExpNode subclasses
// IntLitNode
// AssignExpNode 
// AssignStmtNode 
// PostIncStmtNode
// PostDecStmtNode
// all DeclNode subclasses
// LValNode->IDNode
// IntTypeNode
// VoidTypeNode

intFn : void (var1 : int, var2 : int) {
    var3 : int;
    var3 = 1;
    var3 = -var1;
    var3++;
    var3--;
    var3 = var1 + var2;
    var3 = var1 - var2;
    var3 = var1 * var2;
    var3 = var1 / var2;
    var3 = (var1 + var2) * (var3 - var1);

    var4 : bool;
    var4 = var1 < var2;
    var4 = var1 <= var2;
    var4 = var1 == var2;
    var4 = var1 >= var2;
    var4 = var1 > var2;
}
