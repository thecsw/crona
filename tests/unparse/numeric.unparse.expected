intFn:void(var1:int, var2:int){
	var3:int;
	var3 = 1;
	var3 = (-var1);
	var3++;
	var3--;
	var3 = (var1 + var2);
	var3 = (var1 - var2);
	var3 = (var1 * var2);
	var3 = (var1 / var2);
	var3 = ((var1 + var2) * (var3 - var1));
	var4:bool;
	var4 = (var1 < var2);
	var4 = (var1 <= var2);
	var4 = (var1 == var2);
	var4 = (var1 >= var2);
	var4 = (var1 > var2);
}
